aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -57,4 +102,21 @@ Tensor _s_poisson_cuda(const Tensor& lambda, Generator* gen) {
   return ret;
 }
 
+Tensor _s_gamma_cuda(const Tensor& alpha, Generator* gen) {
+  Tensor ret = alpha.type().tensor(alpha.sizes());
+  auto alpha_ = alpha.toType(ScalarType::Float);
+  AT_DISPATCH_FLOATING_TYPES(ret.type(), "gamma", [&] {
```

Comment:
This does not deploy half type (which explains how stuff like `alpha*alpha` compiles, without THCHalfAutoNumerics.h arithmetic on halfs is not defined)

Reply:
So which would I want?